### DISCUSSION TITLE
/oo/this-super.html

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#31] [Announcements]
# /oo/this-super.html

一份通俗易懂、风趣幽默的Java学习指南，内容涵盖Java基础、Java并发编程、Java虚拟机、Java企业级开发、Java面试等核心知识点。学Java，就认准Java程序员进阶之路

https://tobebetterjavaer.com/oo/this-super.html

**Comment by DylanLJC** [comment]
ThisAsConstrutorParam param;
    Data(ThisAsConstrutorParam param) {
        this.param = param;
    }
这是定义了一个引用类型么？ ThisAsConstrutorParam 这里就相当于int？  Java中定义一个引用类型有什么用？

**Comment by sssnlc** [comment]
“每当创建一个子类对象的时候，也会隐式的创建父类对象，由 super 关键字引用。”
这里说的是不是有问题，我就在想要是，每创建一个子类对象都要把父类对象创建了，那多浪费内存，查了下，
可以看看这篇文章：https://blog.csdn.net/caoruntao_/article/details/82527518
